.github/workflows/ci.yml

```diff
@@ -29,11 +33,12 @@ jobs:
     - name: Setup dotnet
       uses: actions/setup-dotnet@v3
 
-    - name: Install dependencies
+    - name: Restore
       run: dotnet restore
 
     - name: Build
       run: dotnet build --configuration Release --no-restore
 
     - name: Test ${{ matrix.version }}
-      run: dotnet test **/bin/**/${{ matrix.version }}/*.Tests.dll --logger:"console;verbosity=detailed"
+      shell: pwsh
+      run: ForEach ($project in (Get-ChildItem -Path test/*.Tests/*.csproj -File)) { dotnet test $project.FullName --framework ${{ matrix.version }} --configuration Release --no-restore --no-build --logger:"console;verbosity=detailed" }
```

Comment:
Do we not want to exclude Process instrumentation and OneCollector exporter when running the unit tests?

Reply:
We could. A bunch of the workflows have `push` and `pull_request` triggers. I didn't add the `push` triggers for Process Instrumentation & OneCollector Exporter workflows so I let them run here for pushes. Any idea why we have those `push` triggers? I feel like we could remove them (because only admins can push into the protected branches) in which case filtering them out would probably be good in here.

One other thing. We could fold code coverage into the main CI now I think. I'm hoping these changes will stabilize it. On the Process & OneCollector ones I just let them do code coverage while testing (instead of having duplicate jobs). I made it so the upload step in those workflows won't fail the job if it blows up since they are required. We could do the same here. Thoughts?

Reply:
@utpilla I just updated with changes:

* Removed the `push` triggers from workflows
* Added code coverage to main CI and removed the dedicated job
* Excluded Instrumentation.Process & Exporter.OneCollector from the main CI tests